File: 2348-number-of-zero-filled-subarrays/2348-number-of-zero-filled-subarrays.java
class Solution {
    public long zeroFilledSubarray(int[] nums) {
        int i = 0, n = nums.length;
        List<Long> li = new ArrayList<Long>();
        
        while(i < n) {
            int j = i, cnt = 0;
            while(j < n && nums[j] == 0) {
                cnt++;
                j++;
            }
            
            if(cnt == 0) {
                i++;
                continue;
            }
            li.add((long) cnt);
            i = j;
        }
        
        long ans = 0;
        for(Long val: li) {
            ans += (val * (val+1)) / 2;
        }
        
        return ans;
    }
}

File: 2348-number-of-zero-filled-subarrays/2348-number-of-zero-filled-subarrays.cpp
class Solution {
public:
    long long zeroFilledSubarray(vector<int>& nums) {
        int i = 0, n = nums.size();
        vector<long long> v;
        
        while(i < n) {
            int j = i, cnt = 0;
            while(j < n and nums[j] == 0) {
                cnt++;
                j++;
            }
            
            if(cnt == 0) {
                i++;
                continue;
            }
            v.push_back(cnt);
            i = j;
        }
        
        long long ans = 0;
        for(int i=0; i<v.size(); i++) {
            ans += (long long) (v[i] * (v[i]+1)) / 2ll;
        }
        
        return ans;
    }
};